### PR TITLE
Upgrade HCANN, HHH and cleanup JBoss Logger exclusions

### PR DESCRIPTION
**Pull request by Sanne**
- https://hibernate.atlassian.net/browse/HSEARCH-1923
 - https://hibernate.atlassian.net/browse/HSEARCH-1924
 - https://hibernate.atlassian.net/browse/HSEARCH-1928

### CODE REVIEW

**Comment by Sanne** [comment]
I'm changing this PR to include two more trivial fixes

**Comment by DavideD** [comment]
@Sanne I'm ok to merge this pull request. Do you need to add something?

**Comment by gunnarmorling** [comment]
For the CR2 update there is a PR from a contributor. Better take that one to provide some happiness :)

File: serialization/avro/pom.xml [hunk 1]
```diff
@@ -29,7 +29,14 @@
             <groupId>org.apache.avro</groupId>
             <artifactId>avro</artifactId>
         </dependency>
-
+        <dependency>
+            <groupId>org.jboss.logging</groupId>
+            <artifactId>jboss-logging</artifactId>
+        </dependency>
+        <dependency>
```
Comment (gunnarmorling) [inline serialization/avro/pom.xml]: "Provided" scope should suffice. Same for the other usage.

Comment (Sanne) [inline serialization/avro/pom.xml]: What benefit does "provided" provide?

Comment (gunnarmorling) [inline serialization/avro/pom.xml]: These annotations are not needed at runtime, thus I'd omit them.

Comment (Sanne) [inline serialization/avro/pom.xml]: that seems to have changed :-/ try it, it will fail if they are missing.

**Pull request closed without merging**